src/components/fields/image-upload/image-manager/image-manager.ts

```diff
@@ -186,13 +186,15 @@ export const ImageManager = (props: IProps) => {
 		const hasNotPrefilledImages = notPrefilledImages.length > 0;
 		const shouldDirty = hasNotPrefilledImages || gotDeleteImages;
 
+		setCurrentFileIds(uploadedImages.map(({ id }) => id));
+
 		setValue(
 			id,
-			uploadedImages.map(({ dataURL, drawingDataURL, name, uploadResponse, addedFrom }) => ({
+			uploadedImages.map(({ id, dataURL, drawingDataURL, name, uploadResponse }) => ({
+				fileId: id,
```

Comment:
@ryhuz just want to be safe, this doesnt break OS logic right? I have checked the submit case logic and see that you guys build the payload manually so it's fine but I am not sure if other parts of the application will be affected

Reply:
have discussed with ruoling and tested this branch with OS implementation. works ok.

if any other problem presents, will take it separately